### DISCUSSION TITLE
Define scopes based in teams

### DISCUSSION THREAD

**Discussion by erik-arcos-ml** [community/community#8620] [Other Feature Feedback, Questions, & Ideas]
In large organizations the access to repositories is managed by using teams, each team has a set of repositories and each user has access to different teams, therefore, having a scope based in teams would be useful to define a specific set of repositories for the search.

**Comment by colin353** [comment]
Not sure if you're already aware, but you can create custom scopes which search a specific set of repositories:

Click on the "custom scopes" button at the bottom of the scopes list:
![image](https://user-images.githubusercontent.com/1048981/145479235-a92a60bf-d098-4ad7-8247-536dd745b64a.png)

Then you can create a custom scope with a set of repositories you commonly use:
![image](https://user-images.githubusercontent.com/1048981/145479309-d9c53115-bde9-4803-be77-ab7b87abcd72.png)

**Reply by Avivhdr** [reply]
The scopes are great.
Their downside is that they are stored in your local storage and can't be reused by other team members,

**Comment by oddsund** [comment]
This feature would greatly improve code search for our organization. My team has 74 repos, while the org in total has over 1500(including archived). Adding each repo manually to that list is not going to work, especially when each dev has to do it. 

Currently, I've got a custom scope which looks like this; `org:myorg  path:/.*(foo|bar).*/`
Where foo or bar is a part of the paths in all of my teams repository. The downside of this solution is that it excludes all files at the root level, and all auxiliary folders at the root level, e.g. docs. Repo-regex would be a bandaid that would work for us, but filtering by team that owns the repo would solve it.

**Comment by aheld** [comment]
This was literally the first feature I looked for.

We have so many repos and have already organized them using github teams.  Its a pain for each use to have to re-create the scope they want.

I'd love to see some integration between the teams and scopes that makes this shared at an org level.

I know it can be done manually, but then every time a team adds a repo everyone will have to update their scopes.

Also, this is useful for triage / SRE / investigation.  We trigger monitors based on teams right now, so being able to search code based on team would also be appreciated.

**Comment by johannes-mathes** [comment]
Sorry, I can't see the scopes in my view of GitHub? Am I blind?

**Reply by look** [reply]
The screenshot was from our Technology Preview, the UX has changed in the final product.

You can still make scopes by creating a search and then clicking the Save button on the search results page:

<img width="141" alt="Screen Shot 2023-05-15 at 11 39 50 AM" src="https://github.com/community/community/assets/10186/1b9620c4-f012-453a-b488-b643aeeecf46">

The saved search will then be available (and autocompleted) when you type `saved:`.

**Reply by github-vincent-therry** [reply]
Unfortunately the save feature fails with long search (1106 char) (My team have access to 40 repositories in our org) and it fails saving the search :(
